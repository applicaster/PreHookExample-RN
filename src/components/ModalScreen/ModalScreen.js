import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'react-native';
import MediaDetailsScreen from '../MediaDetailsScreen';
import WritePostScreen from '../WritePostScreen';

class ModalScreen extends Component {
  constructor(props) {
    super(props);
    this.closeModal = this.closeModal.bind(this);
  }

  getAnimationType() {
    const { modalName } = this.props;
    switch (modalName) {
      case 'MediaModal':
        return 'fade';
      case 'ReplyToTweetModal':
      case 'WritePostModal':
      default:
        return 'slide';
    }
  }

  closeModal() {
    const { modalName, toggleModal } = this.props;
    toggleModal({ modal: modalName });
  }

  renderModalContent() {
    const { isVisible, modalName } = this.props;
    if (!isVisible) return null;

    switch (modalName) {
      case 'WritePostModal':
        return <WritePostScreen />;
      case 'MediaModal':
        return <MediaDetailsScreen />;
      default:
        return null;
    }
  }

  render() {
    const { isVisible } = this.props;
    
    return (
      <Modal
        animationType={this.getAnimationType()}
        transparent={false}
        visible={isVisible}
        onRequestClose={this.closeModal}
      >
        {this.renderModalContent()}
      </Modal>
    );
  }
}

ModalScreen.propTypes = {
  modalName: PropTypes.string,
  toggleModal: PropTypes.func,
  isVisible: PropTypes.bool,
};

export default ModalScreen;
