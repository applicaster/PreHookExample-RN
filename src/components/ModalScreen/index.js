import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { toggleModal } from '../../actions';
import {
  getModalVisibility,
  getActiveModalName,
} from '../../selectors';

import ModalScreen from './ModalScreen';

const mapStateToProps = state => ({
  isVisible: getModalVisibility(state),
  modalName: getActiveModalName(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ModalScreen);
