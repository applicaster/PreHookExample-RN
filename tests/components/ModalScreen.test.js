import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ModalScreen from '../../src/components/ModalScreen/ModalScreen';

jest.mock('../../src/components/WritePostScreen', () => 'WritePostScreen');
jest.mock('../../src/components/ReplyToTweetScreen', () => 'ReplyToTweetScreen');
jest.mock('../../src/components/MediaDetailsScreen', () => 'MediaDetailsScreen');

describe('ModalScreen', () => {
  test('component renders correctly when it is visible', () => {
    const html = renderer.create(
      <ModalScreen isVisible />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is invisible', () => {
    const html = renderer.create(
      <ModalScreen isVisible={false} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is showing MediaScreen', () => {
    const html = renderer.create(
      <ModalScreen isVisible={false} modalName={'MediaModal'} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is showing WritePostScreen', () => {
    const html = renderer.create(
      <ModalScreen isVisible={false} modalName={'WritePostScreen'} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is showing ReplyToTweetScreen', () => {
    const html = renderer.create(
      <ModalScreen isVisible={false} modalName={'ReplyToTweetScreen'} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
