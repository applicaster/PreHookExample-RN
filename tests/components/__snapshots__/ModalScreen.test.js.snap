// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalScreen component renders correctly when it is invisible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={false}
/>
`;

exports[`ModalScreen component renders correctly when it is showing MediaScreen 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={false}
/>
`;

exports[`ModalScreen component renders correctly when it is showing ReplyToTweetScreen 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={false}
/>
`;

exports[`ModalScreen component renders correctly when it is showing WritePostScreen 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={false}
/>
`;

exports[`ModalScreen component renders correctly when it is visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
/>
`;
